internal/controller/storageclient_controller.go

```diff
@@ -235,7 +235,19 @@ func (r *StorageClientReconciler) reconcilePhases() (ctrl.Result, error) {
 					return reconcile.Result{}, fmt.Errorf("failed to reconcile cephConnection: %v", err)
 				}
 			}
+		case "Subscription":
+			data := map[string]string{}
+			err := json.Unmarshal(eResource.Data, &data)
+			if err != nil {
+				return reconcile.Result{}, fmt.Errorf("failed to unmarshal ConfigMap data: %v", err)
+			}
+
+			utils.AddAnnotation(r.storageClient, utils.DesiredSubscriptionChannelAnnotationKey, data["ocs-operator"])
+			if err := r.update(r.storageClient); err != nil {
+				return reconcile.Result{}, fmt.Errorf("failed to update StorageClient with subscription channel annotation: %v", err)
+			}
```

Comment:
This change around the subscription notification mechanism is a different concern and should be moved to its own PR